consensus/validation_test.go

```diff
@@ -1108,3 +1110,221 @@ func TestValidateBlock(t *testing.T) {
+		}
+		sau = ApplyBlock(vc, b)
+		vc = sau.Context
+		sau.UpdateElementProof(&fce.StateElement)
+	}
+
+	revisions := []types.FileContractRevision{
+		{Parent: fce, Revision: fce.FileContract},
+		{Parent: fce, Revision: fce.FileContract},
+		{Parent: fce, Revision: fce.FileContract},
+	}
+	revisions[0].Revision.RevisionNumber++
+	revisions[0].Revision.ValidRenterOutput.Value = revisions[0].Revision.ValidRenterOutput.Value.Sub(types.Siacoins(1))
+	revisions[0].Revision.MissedRenterOutput.Value = revisions[0].Revision.MissedRenterOutput.Value.Sub(types.Siacoins(1))
+	revisions[0].Revision.ValidHostOutput.Value = revisions[0].Revision.ValidHostOutput.Value.Add(types.Siacoins(1))
+	revisions[0].Revision.MissedHostOutput.Value = revisions[0].Revision.MissedHostOutput.Value.Add(types.Siacoins(1))
+	signRevision(&revisions[0].Revision)
+
+	revisions[1].Revision = revisions[0].Revision
+	revisions[1].Revision.RevisionNumber++
+	revisions[1].Revision.ValidRenterOutput.Value = revisions[1].Revision.ValidRenterOutput.Value.Sub(types.Siacoins(1))
+	revisions[1].Revision.MissedRenterOutput.Value = revisions[1].Revision.MissedRenterOutput.Value.Sub(types.Siacoins(1))
+	revisions[1].Revision.ValidHostOutput.Value = revisions[1].Revision.ValidHostOutput.Value.Add(types.Siacoins(1))
+	revisions[1].Revision.MissedHostOutput.Value = revisions[1].Revision.MissedHostOutput.Value.Sub(types.Siacoins(1))
+	signRevision(&revisions[1].Revision)
+
+	revisions[2].Revision = revisions[1].Revision
+	revisions[2].Revision.RevisionNumber++
+	revisions[2].Revision.ValidRenterOutput.Value = revisions[2].Revision.ValidRenterOutput.Value.Sub(types.Siacoins(1))
+	revisions[2].Revision.MissedRenterOutput.Value = revisions[2].Revision.MissedRenterOutput.Value.Sub(types.Siacoins(1))
+	revisions[2].Revision.ValidHostOutput.Value = revisions[2].Revision.ValidHostOutput.Value.Add(types.Siacoins(1))
+	revisions[2].Revision.MissedHostOutput.Value = revisions[2].Revision.MissedHostOutput.Value.Sub(types.Siacoins(1))
+	signRevision(&revisions[2].Revision)
+
+	// test that a block is invalid if multiple revisions occur in the same block.
+	{
+		revisionTransactionTests := [][]types.Transaction{
+			{{FileContractRevisions: revisions}},
+			{
+				{FileContractRevisions: []types.FileContractRevision{revisions[0]}},
+				{FileContractRevisions: []types.FileContractRevision{revisions[1]}},
+			},
+			{
+				{FileContractRevisions: []types.FileContractRevision{revisions[1]}},
+				{FileContractRevisions: []types.FileContractRevision{revisions[0]}},
+			},
+		}
+		for _, set := range revisionTransactionTests {
+			if err := vc.ValidateBlock(mineBlock(vc, b, set...)); err == nil {
+				t.Fatal("expected invalid block error")
+			} else if !strings.Contains(err.Error(), "multiple times (previously updated in transaction") { // TODO: use errors.Is?
+				t.Fatal("expected multiple update error")
+			}
+		}
+	}
+
+	// test that a block will not be accepted if a revision is signed incorrectly
```

Comment:
`TestValidateTransaction` and `TestValidateTransactionSet` should already cover these sorts of cases; for this PR I only really care about testing a block with multiple revisions and/or resolutions of the same contract